File: Chad/C++/scanf.cpp
#include <iostream>
#include <string>
#include <sstream>

int my_scanf_int() {
    std::string input;
    std::cout << "Inserisci un numero intero:" << std::endl;
    std::getline(std::cin, input);
    std::stringstream ss(input);
    int numero;
    if (ss >> numero) {
        return numero;
    } else {
        std::cerr << "Errore durante la lettura." << std::endl;
        return 0; // O un altro valore per indicare l'errore
    }
}

int main() {
    int numero_inserito = my_scanf_int();
    std::cout << "Hai inserito: " << numero_inserito << std::endl;
    return 0;
}

File: Chad/scanf.rs
use std::io;

fn my_scanf_int() -> Result<i32, String> {
    let mut input = String::new();
    io::stdin()
        .read_line(&mut input)
        .map_err(|e| e.to_string())?;

    input.trim().parse::<i32>().map_err(|e| e.to_string())
}

fn main() -> Result<(), String> {
    println!("Inserisci un numero intero:");
    match my_scanf_int() {
        Ok(numero) => println!("Hai inserito: {}", numero),
        Err(errore) => println!("Errore durante la lettura: {}", errore),
    }
    Ok(())
}
